community/values/src/main/java/org/neo4j/values/storable/PointValue.java

```diff
@@ -264,153 +261,64 @@ public boolean withinRange( PointValue lower, boolean includeLower, PointValue u
 
     public static PointValue fromMap( MapValue map )
     {
-        AnyValue[] fields = new Value[PointValueField.values().length];
-        for ( PointValueField f : PointValueField.values() )
+        PointCSVHeaderInformation fields = new PointCSVHeaderInformation();
+        for ( Map.Entry<String,AnyValue> entry : map.entrySet() )
         {
-            AnyValue fieldValue = map.get( f.name().toLowerCase() );
-            fields[f.ordinal()] = fieldValue != Values.NO_VALUE ? fieldValue : null;
+            fields.assign( entry.getKey().toLowerCase(), entry.getValue() );
```

Comment:
Before, you had some converting between `NO_VALUE` and `null` here. Can we never get `NO_VALUE` in the current solution?

Reply:
We might get a ```NO_VALUE```, but ```assign()``` would throw an exception if we encounter it (would be the same if we pass in null, so it makes no difference to check for ```NO_VALUE```